test/controller_test.rb

```diff
@@ -282,4 +282,27 @@ class PostsControllerTest < ActionDispatch::IntegrationTest
     json = JSON.parse(@response.body)
     assert_equal [post1.id], (json.map { |post| post.fetch("id") })
   end
+
+  test "it filters on metadata by jsonb key" do
+    post = Post.create!(metadata: { 'a' => 4 }.to_json)
+    Post.create!(metadata: { 'b' => 5 }.to_json)
+
+    # Stubs are needed because the dummy app DB is not PostgreSQL
+    instance_mock = Minitest::Mock.new
+    instance_mock.expect :call, Post.where(id: post.id), [Post.all, Hash, ActionController::Parameters, Array]
+
+    class_mock = Minitest::Mock.new
+    class_mock.expect :call, instance_mock, [Sift::Parameter]
+
+    Sift::WhereHandler.stub :new, class_mock, [Sift::Parameter] do
+      get("/posts", params: { filters: { metadata: { 'a' => 4 }.to_json } })
```

Comment:
nb: can we add a spec to make sure the array case works as expected? it does not look like that is covered anywhere

Reply:
thanks, unit spec added